app/src/main/java/com/example/drawingApp/ColorPickerUtility.kt

```diff
@@ -0,0 +1,222 @@
+         * update your color afterward
+         */
+        onSatOrValUpdate: (Float, Float) -> Unit,
+        /**
+         * Called whenever text needs to be updated, meaning on any change, hsv, rgb, or text.
+         * Used to give the information of hsv and previous color.
+         * if there was no previous color pass it black. This is used to help
+         * populate the sheets text views and update the slider and gradient indicators as well as set the previous color box.
+         */
+        onColorTextForceUpdate: () -> DrawingViewModel.ColorPack,
+        /**
+         * Called whenever one of the RGB texts is changed by focus by a user.
+         * it passes a ColorInt as the first parameter, used to decide what value to set.
+         * the second int is the actual value you will set your R,G,or B value to.
+         * Color.RED corresponds to red,Color.GREEN to green, BLUE to blue
+         */
+        onRGBTextChange: (Int, Int) -> Unit,
+        /**
+         * called whenever the dialog is dismissed or submitted
+         * on actual submission the color will not be null so take it in and set it wherever you need the selected color and set the hsv and rgb fields to defaults.
+         * on null don't do anything with it, don't set rgb and hsv to defaults.
+         */
+        onSubmission: (Color?) -> Unit
+    ) {
+        val sheet = BottomSheetDialog(context)
+        val inflater = LayoutInflater.from(context)
+        val view = inflater.inflate(R.layout.color_picker_bottom_sheet, null)
+        val submit = view.findViewById<TextView>(R.id.dialogSubmit)
+        val cancel = view.findViewById<TextView>(R.id.dialogCancel)
+        val hView = view.findViewById<ForceEditText>(R.id.hslH)
+        val sView = view.findViewById<ForceEditText>(R.id.hslS)
+        val vView = view.findViewById<ForceEditText>(R.id.hslL)
+        val rView = view.findViewById<ForceEditText>(R.id.rgbR)
+        val gView = view.findViewById<ForceEditText>(R.id.rgbG)
+        val bView = view.findViewById<ForceEditText>(R.id.rgbB)
+        val chView = view.findViewById<ForceEditText>(R.id.colorHex)
+        val slider = view.findViewById<ColorSlider>(R.id.slider)
+        val square = view.findViewById<GradientSquare>(R.id.gradientSquare)
+        val activeBox = view.findViewById<TextView>(R.id.colorOne)
+        val previousBox = view.findViewById<TextView>(R.id.colorTwo)
+        val previousColor = onColorTextForceUpdate().previousColor
+        previousBox.setBackgroundColor(previousColor.toArgb())
+        //used to prevent dragging when clicking on the gradient or the slider
+        val onViewTouch = {
+            sheet.behavior.isDraggable = false
+        }
+        val onViewLift = {
+            sheet.behavior.isDraggable = true
+        }
+        //used to set the color of the active box
+        val setActiveBox: (FloatArray) -> Unit = { hsv ->
```

Comment:
this is a lambda and should start with "on", ex "onSetActiveBox" in this case

Reply:
I'll be pretty honest and say the routing with this one does confuse me a little

so from what I can follow you are passing this lambda as an argument in an invoked lambda (that is called whenever you change something in one of the views). This lambda then updates all of the views in this scope.

I feel like making this a local function could also make this clearer? I am unsure if that approach is better/more professional here, but the way you reset your views feels like it goes through some extra complexity here.

It might be the case where this makes it more reusable? with a sheet like this, I feel like this view is already specialized and improving the readability here should take priority, but Nic will have better insight for this.

Reply:
this lambda is esssentialy a local function, rather than giving a function signature and having to pass the function the views and all that it seemed better to just acknowledge this was only going to be used in this scope and to just create it as a lambda to be used in the function.